src/tracks/InstrumentTrack.cpp

```diff
@@ -938,6 +938,21 @@ void InstrumentTrack::setPreviewMode( const bool value )
 
 
 
+void InstrumentTrack::replaceInstrument(DataFile dataFile)
+{
+	int effectChannel = effectChannelModel()->value();
```

Comment:
Would suggest adding a small comment explaining why we're doing this. Maybe something like:
`// save FX channel so it doesn't get overwritten by loadSettings`

Reply:
```suggestion
	// loadSettings clears the FX channel, so we save it here and set it back later
	int effectChannel = effectChannelModel()->value();
```

Reply:
Good call! Fixed :)